### ISSUE TITLE
Train First Speaker Verification Model

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/SimilarityLearning#19]
Use SpeakerModel and VoxCeleb1 to train a first speaker verification model.
Cross entropy takes priority over the rest of the losses as discussed in previous meetings.
Make sure to use EER as the validation metric.

**Comment by juanmc2005** [comment]
Cannot directly use VoxCeleb1 data with SincNet. We need to split the samples in chunks, like they explain in the paper (at least at the beginning).
This missing chunks are most likely causing an out of memory issue.

@hbredin any way of chunking audio segments already implemented in pyannote?

**Comment by hbredin** [comment]
Not sure what you need.

`SpeechSegmentGenerator` already yields audio chunks, whose duration is controlled by the `duration` parameter.

Can you clarify your needs?

**Comment by juanmc2005** [comment]
Nevermind, I misunderstood the 
`duration` parameter. I will try changing that on Monday.
Thanks!

**Comment by juanmc2005** [comment]
@hbredin Please tell me if I can do something to help solve the 3199 issue.
If you want to force it to see what it looks like, you can use the `sv-train2` branch and run:
```python -W ignore main.py --task speaker --loss softmax --epochs 1500 --no-plot --no-save --batch-size 100 --log-interval 5```
The problem occurs around 25% of the first epoch

**Comment by hbredin** [comment]
I am not sure why this happens.

One way of understanding this behavior is to create a simple script that simply iterates forever on SpeechSegmentGenerator and stops as soon as the number of samples is not 3200.

You can edit SpeechSegmentGenerator temporarily so that it also returns the value of `sub_segment` and `files[i]`.

**Comment by hbredin** [comment]
Starting from here, we will be able to investigate what is happening

**Comment by juanmc2005** [comment]
Got it. I'm switching to STS for the time being, to integrate the model and dataset.
After that I'll start working this out.
Thanks!

**Comment by juanmc2005** [comment]
Will unblock and use 0-padding for samples with wrong dimensions while we still look for what's causing this problem.

**Comment by juanmc2005** [comment]
The validation code is too expensive to run after each epoch, will use a separate script to run validations in parallel when a model is saved. This will be done as part of another issue: #25 
The remaining task for this issue is being able to train a model without validation.

**Comment by juanmc2005** [comment]
**Update:** Validation can be done in-training for VoxCeleb1, but it would be useful to parallelize anyway to tackle VoxCeleb2.
Models can be trained using cross entropy, although the results are not very good. Will close this issue and open another one to address that problem.
